velox/dwio/parquet/reader/RepeatedColumnReader.cpp

```diff
@@ -108,21 +108,23 @@ void ensureRepDefs(
 }
 
 MapColumnReader::MapColumnReader(
-    std::shared_ptr<const dwio::common::TypeWithId> requestedType,
+    const std::shared_ptr<const dwio::common::TypeWithId>& requestedType,
+    const std::shared_ptr<const dwio::common::TypeWithId>& dataType,
     ParquetParams& params,
     common::ScanSpec& scanSpec)
     : dwio::common::SelectiveMapColumnReader(
           requestedType,
-          requestedType,
+          dataType,
           params,
           scanSpec) {
+  DWIO_ENSURE_EQ(fileType_->id(), dataType->id(), "working on the same node");
```

Comment:
I suggest to rename fileType_ and the dataType parameter name to either "dataFileType" or "fileType" so that the code is easier to understand. You see, fileType_ comes from dataType parameter but they are named differently, making the code not as straitforward as it should be.

Reply:
We can rename it to `fileType_` in the future.  For now `dataType` and `fileType` are used interchangeably, and `requestedType` is different, the situation is not too bad.